lib/pos_consensus.go

```diff
@@ -49,6 +49,22 @@ func (cc *FastHotStuffConsensus) Start() error {
 	// Fetch the current tip of the chain
 	tipBlock := cc.blockchain.BlockTip()
 
+	// If the chain is not at the final PoW block height or higher, then we cannot start the PoS consensus.
+	if tipBlock.Height < cc.blockchain.params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight-1 {
+		return errors.Errorf(
+			"FastHotStuffConsensus.Start: Block tip %d is not at the final PoW block height", tipBlock.Height,
+		)
+	}
+
+	// If the consensus is starting for the first time and is at the final PoW block height, then we need
+	// to inject the genesis blocks.
+	if err := cc.tryInjectGenesisBlocks(); err != nil {
+		return errors.Errorf("Error injecting genesis blocks: %v", err)
+	}
+
+	// Fetch the current tip of the chain
+	tipBlock = cc.blockchain.BlockTip()
```

Comment:
Ahh okay I was confused why we were getting the tipBlock again. Can we add a comment that we fetch the tip again so we have the tip after injecting the PoS genesis blocks?